### DISCUSSION TITLE
Using polymorphic links

### DISCUSSION THREAD

**Discussion by ghost** [geldata/gel#5768] [Q&A] [answered]
Hello

I am trying to model a hierarchical structure and I am having issues with the polymorphic links and insert

I have replicated the issue with a dummy minimal schema
```
module default {

    abstract type Field {
        required name: str;
    }

    type StringField  extending Field {
        value: str;
    }

    type IntField  extending Field {
        value: int64;
    }

    type CompositeField  extending Field {
        multi value: Field {
            constraint exclusive;
        }
    }
}

```

then trying to insert a nested field like:

```
insert CompositeField {
  name := <str>'CompositeField1',
  value :=  {   
           ( 
            insert StringField {
              name := <str>'StringField1',
              value := <str>"XXXXX"
            }
          ),
          ( 
            insert IntField {
              name := <str>'IntField1',
              value := <int64>666
            }
          ),
          
  }
};
```

Is giving me the following error :
SchemaError: cannot create a union of std::int64, std::str
Any insertions with the same type , works, but mixed type always gibe the error.

It seems that I cannot have several derived types with the same name (value) property with different types. 
it has to be a different name.

so the schema needs to be
```
module default {

    abstract type Field {
        required name: str;
    }

    type StringField  extending Field {
        s_value: str;
    }

    type IntField  extending Field {
        i64_value: int64;
    }

    type CompositeField  extending Field {
        multi fields: Field {
            constraint exclusive;
        }
    }
}

```
thing that is overcomplicating evening for me..

So the question is , is there a way to be able to have the same name with different type in derived types?
 
Regards
Alberto.

**Comment by raddevon** [comment, marked as answer]
I checked into this for you. It's not currently possible and will not be in the future because we are unable to represent a union of `str` and `int64`.

If you need this sort of flexibility over strict typing, one solution could be to make `value` a `json` type property. If you want to prioritize strict typing instead, your solution of naming properties differently might be best.

**Reply by ghost** [reply]
Thank you for you response.
